lib/utils/middlewares/path_param.ts

```diff
@@ -0,0 +1,29 @@
+import { left, right } from "../either";
+
+import { IRequestMiddleware } from "../request_middleware";
+import { IResponseErrorValidation, ResponseErrorValidation } from "../response";
+
+/**
+ * A request middleware that validates the presence of a valid `param` parameter
+ * in the request. In case the parameter is missing or is not valid, the middleware
+ * returns an `IResponseErrorValidation`.
+ */
+export const PathParamMiddleware: <T>(
+  paramName: string,
+  // tslint:disable-next-line:no-any
+  testfn: (arg: any) => arg is T
```

Comment:
I think you just need a `(string) => boolean` here

Reply:
maybe it's clearer with the current signature

Reply:
why ask for an `any`, when you know you're going to pass a `string`? always go for the narrower type, avoid asking for an `any` at all costs

also, you don't need a type guard here, a function that simply returns a `boolean` is sufficient (that doesn't mean you can't still pass a type guard here) - the principle is to make the types and parameter as few and as more specific as possibile